pkg/config/config.go

```diff
@@ -521,19 +521,37 @@ func LoadWithoutSecret() error {
 	return load(Datadog, "datadog.yaml", false)
 }
 
-func load(config Config, origin string, loadSecret bool) error {
-	if err := config.ReadInConfig(); err != nil {
-		log.Warnf("Error loading config: %v", err)
-		return err
-	}
-
+func checkForUnknownKeys(config Config) bool {
+	hasUnknownKeys := false
 	knownKeys := config.GetKnownKeys()
 	loadedKeys := config.AllKeys()
 	for _, key := range loadedKeys {
 		if _, found := knownKeys[key]; !found {
-			log.Warnf("Unknown key in config file: %v", key)
+			// Check if any subkey terminated with a '.*' wildcard is marked as known
+			// e.g.: apm_config.* would match all sub-keys of apm_config
+			splitPath := strings.Split(key, ".")
+			for j := range splitPath {
+				subKey := strings.Join(splitPath[:j+1], ".") + ".*"
+				if _, found = knownKeys[subKey]; found {
+					break
+				}
+			}
+			if !found {
+				log.Warnf("Unknown key in config file: %v", key)
+				hasUnknownKeys = true
+			}
 		}
 	}
+	return hasUnknownKeys
+}
+
+func load(config Config, origin string, loadSecret bool) error {
+	if err := config.ReadInConfig(); err != nil {
+		log.Warnf("Error loading config: %v", err)
+		return err
+	}
+
+	checkForUnknownKeys(config)
```

Comment:
nit: I think it would be clearer to use the returned bool here to decide if we should log something instead of taking an action in `checkForUnknownKeys`

Reply:
Indeed. Done!